core/modules/wcontrol/SqlConnMgr.cc

```diff
@@ -0,0 +1,82 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2020 LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "wcontrol/SqlConnMgr.h"
+
+#include "lsst/log/Log.h"
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.wcontrol.SqlConnMgr");
+}
+
+using namespace std;
+
+namespace lsst {
+namespace qserv {
+namespace wcontrol {
+
+
+void SqlConnMgr::_take(bool scanQuery) {
+    ++_totalCount;
+    std::unique_lock<std::mutex> uLock(_mtx);
+    _tCv.wait(uLock, [this, scanQuery](){
+        bool ok = _sqlConnCount < _maxScanSqlConnections;
+        if (not scanQuery) {
+            ok = _sqlConnCount < _maxSqlConnections;
+        }
+        return ok;
+    });
+    ++_sqlConnCount;
+}
+
```

Comment:
Missing empty line. There should be 2 lines between methods in the implementation files.